Tasks/cURLUploader/task.json

```diff
@@ -35,28 +35,59 @@
             "helpMarkDown": "File(s) to be uploaded. Wildcards can be used. For example, `**\\*.zip` for all ZIP files in all subfolders."
         },
         {
+            "name": "authType",
+            "type": "pickList",
+            "label": "Authentication Method",
+            "defaultValue": "ServiceEndpoint",
+            "helpMarkDown": "",
+            "options": {
+                "ServiceEndpoint": "Service Endpoint",
+                "UserAndPass": "Username and Password"
+            }
+        },
+        {
+            "name": "serviceEndpoint",
+            "type": "connectedService:Generic",
+            "label": "Service Endpoint",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "The service endpoint with the credentials for the server authenication.",
+            "visibleRule": "authType = ServiceEndpoint"
+        },
+        {
             "name": "username",
             "type": "string",
             "label": "Username",
             "defaultValue": "",
             "required": false,
-            "helpMarkDown": "Specify the username for server authentication."
+            "helpMarkDown": "Specify the username for server authentication.",
+            "visibleRule": "authType = UserAndPass"
         },
         {
             "name": "password",
             "type": "string",
             "label": "Password",
             "defaultValue": "",
             "required": false,
-            "helpMarkDown": "Specify the password for server authentication. Use a new build variable with its lock enabled on the Variables tab to encrypt this value."
+            "helpMarkDown": "Specify the password for server authentication. Use a new build variable with its lock enabled on the Variables tab to encrypt this value.",
+            "visibleRule": "authType = UserAndPass"
         },
         {
             "name": "url",
             "type": "string",
             "label": "URL",
             "defaultValue": "",
             "required": true,
-            "helpMarkDown": "Specify the URL to where the file(s) will be uploaded. The directory should end with a trailing slash. Possible URL protocols include `DICT://`, `FILE://`, `FTP://`, `FTPS://`, `GOPHER://`, `HTTP://`, `HTTPS://`, `IMAP://`, `IMAPS://`, `LDAP://`, `LDAPS://`, `POP3://`, `POP3S://`, `RTMP://`, `RTSP://`, `SCP://`, `SFTP://`, `SMTP://`, `SMTPS://`, `TELNET://` and `TFTP://`."
+            "helpMarkDown": "Specify the URL to where the file(s) will be uploaded. The directory should end with a trailing slash. Possible URL protocols include `DICT://`, `FILE://`, `FTP://`, `FTPS://`, `GOPHER://`, `HTTP://`, `HTTPS://`, `IMAP://`, `IMAPS://`, `LDAP://`, `LDAPS://`, `POP3://`, `POP3S://`, `RTMP://`, `RTSP://`, `SCP://`, `SFTP://`, `SMTP://`, `SMTPS://`, `TELNET://` and `TFTP://`.",
```

Comment:
Can you please verify that the "Generic" service endpoint allows non-HTTP URLs like `SCP://`?